bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -0,0 +1,325 @@
+            0
+        }
+    }
+}
+
+pub mod tests {
+    use super::*;
+    use crate::raw::{config::*, error::Error};
+    use crate::testing::Pair;
+
+    const SAMPLES: usize = 100;
+
+    #[allow(unused_variables)]
+    struct HostVerifyData {
+        callback_invoked: bool,
+        allow: bool,
+        host: &'static str,
+    }
+    struct CertKeyPair {
+        cert: &'static [u8],
+        key: &'static [u8],
+    }
+    #[allow(unused_variables, dead_code)]
+    impl CertKeyPair {
+        fn default() -> Self {
+            CertKeyPair {
+                cert: include_str!("../../../../../tests/pems/rsa_4096_sha512_client_cert.pem")
+                    .as_bytes(),
+                key: include_str!("../../../../../tests/pems/rsa_4096_sha512_client_key.pem")
+                    .as_bytes(),
+            }
+        }
+        fn ecdsa() -> Self {
+            CertKeyPair {
+                cert: include_str!("../../../../../tests/pems/ecdsa_p256_pkcs1_cert.pem")
+                    .as_bytes(),
+                key: include_str!("../../../../../tests/pems/ecdsa_p256_pkcs1_key.pem")
+                    .as_bytes(),
+            }
+        }
+        fn cert(&mut self) -> &'static [u8] {
+            self.cert
+        }
+        fn key(&mut self) -> &'static [u8] {
+            self.key
+        }
+    }
+
+    pub fn build_config(cipher_prefs: &str) -> Result<crate::raw::config::Config, Error> {
+        let mut builder = Builder::new();
+        let mut keypair = CertKeyPair::default();
+        // Build a config
+        builder
+            .set_cipher_preference(cipher_prefs)
+            .expect("Unable to set config cipher preferences");
+        builder
+            .load_pem(keypair.cert(), keypair.key())
+            .expect("Unable to load cert/pem");
+        unsafe {
+            let mut ctx = HostVerifyData {
+                allow: true,
+                callback_invoked: false,
+                host: "127.0.0.1",
+            };
+            let ctx_ptr: *mut core::ffi::c_void = &mut ctx as *mut _ as *mut core::ffi::c_void;
+            builder
+                .set_verify_host_callback(Some(verify_host_cb), ctx_ptr)
+                .expect("Unable to set a host verify callback.");
+        }
+        builder
+            .disable_x509_verification()
+            .expect("Unable to disable x509 verification");
+        Ok(builder.build().expect("Unable to build server config"))
+    }
+
+    // host verify callback for x509
+    // see: https://github.com/aws/s2n-tls/blob/main/docs/USAGE-GUIDE.md#s2n_verify_host_fn
+    unsafe extern "C" fn verify_host_cb(
+        hostname: *const i8,
+        hostname_len: usize,
+        context: *mut core::ffi::c_void,
+    ) -> u8 {
+        let host_str = ::std::str::from_utf8(::std::slice::from_raw_parts(
+            hostname as *const u8,
+            hostname_len,
+        ));
+        let _ctx: &mut HostVerifyData = &mut *(context as *mut HostVerifyData);
```

Comment:
Ah this is why it _currently_ doesn't behave incorrectly... You're not actually using the context. Do you plan on doing this at some point? It might be better to just pass void as the context since you're just accepting everything.

Reply:
I'll revert back to void, which was the initial setup; my attempt to emulate how the unit tests use context doesn't seem valid here.